packages/cli-kit/src/common/object.ts

```diff
@@ -0,0 +1,13 @@
+import deepMerge from 'deepmerge'
+
+/**
+ * Deep merges the two objects and returns a new object
+ * with the merge result.
+ * @param lhs {any} One of the objects to be merged.
+ * @param rhs {any} Another object to be merged.
+ * @returns {any} A Javascrip tobject with th emerged objects.
+ */
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export function deepMergeObjects(lhs: any, rhs: any): any {
```

Comment:
can `typescript` do its magic and infer the result?
@shauns you are the expert in types, would something like this be possible?
```ts
deepMergeObjects<TLhs, TRhs>(lhs: TLhs, rhs: TRhs): TLhs & TRhs
```

Reply:
Actually, deepmerge is already typed... [link](https://github.com/TehShrike/deepmerge/blob/master/index.d.ts#L2)
Can we avoid using `any` here?